receiver/splunkhecrawreceiver/receiver.go

```diff
@@ -0,0 +1,263 @@
+	r.resourceCustomizer = r.createResourceCustomizer()
+
+	go func() {
+		if errHTTP := r.server.Serve(ln); errHTTP != http.ErrServerClosed {
+			host.ReportFatalError(errHTTP)
+		}
+	}()
+
+	return err
+}
+
+// Shutdown tells the receiver that should stop reception,
+// giving it a chance to perform any necessary clean-up.
+func (r *splunkReceiver) Shutdown(context.Context) error {
+	return r.server.Close()
+}
+
+func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	if r.logsConsumer == nil {
+		ctx = r.obsrecv.StartMetricsOp(ctx)
+	}
+	reqPath := req.URL.Path
+	if !r.config.pathGlob.Match(reqPath) {
+		r.failRequest(ctx, resp, http.StatusNotFound, notFoundRespBody, nil)
+		return
+	}
+
+	if req.Method != http.MethodPost {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBody, nil)
+		return
+	}
+
+	encoding := req.Header.Get(httpContentEncodingHeader)
+	if encoding != "" && encoding != gzipEncoding {
+		r.failRequest(ctx, resp, http.StatusUnsupportedMediaType, invalidEncodingRespBody, nil)
+		return
+	}
+
+	bodyReader := req.Body
+	if encoding == gzipEncoding {
+		var err error
+		bodyReader, err = gzip.NewReader(bodyReader)
+		if err != nil {
+			r.failRequest(ctx, resp, http.StatusBadRequest, errGzipReaderRespBody, err)
+			return
+		}
+	}
+
+	if req.ContentLength == 0 {
+		resp.Write(okRespBody)
+		return
+	}
+
+	sc := bufio.NewScanner(bodyReader)
+
+	ld := pdata.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	r.resourceCustomizer(req, rl.Resource())
+	ill := rl.InstrumentationLibraryLogs().AppendEmpty()
+
+	for sc.Scan() {
+		logRecord := ill.Logs().AppendEmpty()
+		logLine := sc.Text()
+		logRecord.Body().SetStringVal(logLine)
+	}
+	decodeErr := r.logsConsumer.ConsumeLogs(ctx, ld)
+
+	if decodeErr != nil {
+		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, decodeErr)
+	} else {
+		resp.WriteHeader(http.StatusAccepted)
+		resp.Write(okRespBody)
+	}
+}
+
+func (r *splunkReceiver) failRequest(
+	ctx context.Context,
+	resp http.ResponseWriter,
+	httpStatusCode int,
+	jsonResponse []byte,
+	err error,
+) {
+	resp.WriteHeader(httpStatusCode)
+	if len(jsonResponse) > 0 {
+		// The response needs to be written as a JSON string.
+		_, writeErr := resp.Write(jsonResponse)
+		if writeErr != nil {
+			r.logger.Warn("Error writing HTTP response message", zap.Error(writeErr))
+		}
+	}
+
+	msg := string(jsonResponse)
+
+	reqSpan := trace.FromContext(ctx)
+	reqSpan.AddAttributes(
+		trace.Int64Attribute(conventions.AttributeHTTPStatusCode, int64(httpStatusCode)),
+		trace.StringAttribute(conventions.AttributeHTTPStatusText, msg))
+	traceStatus := trace.Status{
+		Code: trace.StatusCodeInvalidArgument,
+	}
+	if httpStatusCode == http.StatusInternalServerError {
+		traceStatus.Code = trace.StatusCodeInternal
+	}
+	if err != nil {
+		traceStatus.Message = err.Error()
+	}
+	reqSpan.SetStatus(traceStatus)
+	reqSpan.End()
```

Comment:
Even if the collector is creating spans, I would not expect code to end a span that it didn't start, this doesn't look correct

Reply:
See above.